kazoo/handlers/utils.py

```diff
@@ -354,7 +354,11 @@ def fileobj_to_fd(fileobj):
             raise TypeError("Invalid file object: " "{!r}".format(fileobj))
     if fd < 0:
         raise TypeError("Invalid file descriptor: {}".format(fd))
-    os.fstat(fd)
+
+    # XXX: needed?
+    # selectors_module.register handle fileobject, do we need the use of fileobj_to_fd()?
+    # is this test valid for Windows?
+    # os.fstat(fd)
```

Comment:
Hello @JetDrag,

Sorry to summon you here. This `os.fstat(fd)` call is generating crashes on Windows because the handle is, it seems, invalid. With this call commented out, the crash does not appear (and tests are running fine).
Since you did this implementation, I was wondering if:
- there was a reason using the `fileobj_to_fd()` method instead of directly giving the fileobj to `selectors_module.register()` (which seems to be capable of handling fileobj directly)?
- you know if the `os.fstat(fd)` test is valid for Windows?

Thank you.

Reply:
I second this.
This `os.fstat()` does not do anything. 
Actually, python3 handles `os.stat()` on FD or file just fine.

Reply:
https://docs.python.org/3/library/os.html#os.stat

Reply:
Hi, @StephenSorriaux @ceache 

The function `fileobj_to_fd` is same as `_fileobj_to_fd` function in `selectors` module  almost except the last second line `os.fstat(fd)`.

I did this implementation to simulate the raw select behavior via test case `test_selectors_select`, which is modified from Python official test cast `test_select`.

If I don't do that, the test cast `test_errno` in the `test_selectors_select` will fail, because the exception (OSError with errno 22, EINVAL) raise by  `selectors._BaseSelectorImpl` module don't meet the test case requirement：raise OSError with errno 9 (EBADF) when select meet a closed fd.

Maybe we cloud make some other simulate implementation to make those spec work on the Windows?

Reply:
https://github.com/python/cpython/blob/main/Lib/test/test_select.py#L32  `test_errno()` for reference.

I do not think we care about this behavior in Kazoo.
Actually, the err conditions with selector when selecting "bad file descriptors" are varied depending on platform and selector implementations (there are threads about this on the Python bug report).
This is here trying to make sure that selectors implementations are all equivalente to `select()` behavior and we really can't do that.

IMHO. the only thing we need to do is ensure how Kazoo reacts to a closed FD.
In other words, be it `EINVAL` or `EBADF` or other, we should raise an uniform Kazoo error (with Python3, we could pass up the original, OS/Implementation dependent Exception).

What do you think?

Reply:
It maybe the best solution.

Reply:
Thank you both for your thoughts. As suggested, I tried to unify the exceptions that can be raised and catch that in the `test_errno()` testcase. That may be too naive, not sure really, so I would love your feedback on this.